apps/frontend/src/components/filters.tsx

```diff
@@ -83,19 +99,42 @@ const Filters = ({ filters, onFilterChange }: FiltersProps) => {
       const priceValue: number = +event.target.value;
       setMaxPrice(priceValue >= 0 ? priceValue : priceValue * -1);
     }
-    setMaxPrice(event.target.value ? Number(event.target.value) : null);
+    setMaxPrice(newMaxPrice === null ? null : +newMaxPrice);
+    onFilterChange({
+      minPrice: minPrice,
+      maxPrice: event.target.value === "" ? null : +event.target.value,
+      status: status,
+      searchType: type,
+      latitude: latitude,
+      longitude: longitude,
+    });
   };
 
   const handleStatusChange = (event: SelectChangeEvent<string>) => {
     setStatus(event.target.value);
+    onFilterChange({
+      minPrice: minPrice,
+      maxPrice: maxPrice,
+      status: event.target.value,
+      searchType: type,
+      latitude: latitude,
+      longitude: longitude,
+    });
   };
 
   const handleTypeChange = (event: SelectChangeEvent<string>) => {
     setType(event.target.value);
+    onFilterChange({
```

Comment:
```
onFilterChange({
...filterState,
searchType: newSearchType
});
```

Reply:
Changed

Reply:
I ended up not changing these because the way java seems to be updating it is not getting the correct values when making multiple updates quickly